UPGRADING.md

```diff
@@ -148,12 +148,20 @@ Additionally, the `appmodule.Environment` interface is introduced to fetch diffe
 This should be used as an alternative to using `sdk.UnwrapContext(ctx)` to fetch the services.
 It needs to be passed into a module at instantiation. 
 
-`x/circuit` is used as an example.:
+`x/circuit` is used as an example:
 
 ```go
 app.CircuitKeeper = circuitkeeper.NewKeeper(runtime.NewEnvironment((keys[circuittypes.StoreKey])), appCodec, authtypes.NewModuleAddress(govtypes.ModuleName).String(), app.AuthKeeper.AddressCodec())
 ```
 
+If your module requires a message server or query server, it should be passed in the environment as well.
+
+```go
```

Comment:
Reminder for myself tomorrow:
- change this to diff here, remove extra space
- fix conflict and use typeurl of proposal msg for can invoke
- fix lint
- fix typo proposed by ai bot